sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            if (str == null)
+            {
+                throw Errors.ArgumentNull(nameof(str));
+            }
+
+            if (str.Length != 9 && str.Length != 10)
+            {
+                throw Errors.InvalidArgument(nameof(str));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            // Set sticky bit
+            if (str.ToLower(CultureInfo.InvariantCulture)[str.Length - 1] == 't')
+            {
+                pathPermissions.StickyBit = true;
+            }
+            else
+            {
+                pathPermissions.StickyBit = false;
+            }
+
+            // Set extended info in ACL
+            if (str.Length == 10)
+            {
+                if (str[9] == '+')
+                {
+                    pathPermissions.ExtendedInfoInAcl = true;
+                }
+                else
+                {
+                    throw Errors.InvalidFormat(nameof(str));
+                }
+            }
+            else
+            {
+                pathPermissions.ExtendedInfoInAcl = false;
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseSymbolic(str.Substring(0, 3), allowStickyBit: false);
+            pathPermissions.Group = RolePermissionsExtensions.ParseSymbolic(str.Substring(3, 3), allowStickyBit: false);
+            pathPermissions.Other = RolePermissionsExtensions.ParseSymbolic(str.Substring(6, 3), allowStickyBit: true);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Returns the octal representation of this PathPermissions as a string.
+        /// </summary>
+        /// <returns>string</returns>
+        public string ToOctalString()
+        {
+            var sb = new StringBuilder();
+
+            if (StickyBit)
+            {
+                sb.Append(1);
+            }
+            else
+            {
+                sb.Append(0);
+            }
+
+            sb.Append(Owner.ToOctalString());
+            sb.Append(Group.ToOctalString());
+            sb.Append(Other.ToOctalString());
+
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Returns the symbolic represenation of this PathPermissions as a string.
+        /// </summary>
+        /// <returns>string</returns>
+        public string ToSymbolicString()
+        {
+            var sb = new StringBuilder();
+            sb.Append(Owner.ToSymbolicString());
+            sb.Append(Group.ToSymbolicString());
+            sb.Append(Other.ToSymbolicString());
+
+            if (StickyBit)
+            {
+                sb.Remove(8, 1);
+                sb.Append("t");
+            }
+
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Equals.
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool Equals(PathPermissions other)
```

Comment:
The Equals methods are only here for the unit tests.  The other possibility would be to add helper methods in the DataLakeTestBase to check PathPermissions equality.

Reply:
I'd rather not have public APIs that only exist for tests - and these are also kind of weird since we're not a struct and not IEquatable.  I'd be nicer to add test helpers and remove these.

Reply:
Removed.